CIP-0021/CIP-0021.md

```diff
@@ -0,0 +1,115 @@
+---
+CIP: 0021
+Title: Restrictions on transactions signed by hardware wallets
```

Comment:
the title is a bit misleading. Rather this CIP wants to establish some best practices for HW wallets.

Reply:
Perhaps the intention is unclear and should be made clearer in the CIP. It does not comment on best practices, but lists limitations arising from hardware limitations of Trezor and especially Ledger.

The intended CIP audience are people integrating with HW wallets (developers of software wallets and similar tools), not HW wallet developers. In other words the CIP should define how you should build your transactions in order for them to be compatible (signable) with HW wallets - specifically Ledger and Trezor (not sure if there are other significant ones currently). In general, the rules should be a superset of restrictions for each HW wallet, so if you follow this CIP, you would be compatible with all HW wallets, and if you don't, you risk that some parties use HW wallets and will not be able to sign your transaction. This is especially important in context of transactions signed by multiple parties.

Reply:
I'd suggest using the term "interoperability" in the title, e.g. "Transaction guidelines for interoperability with hardware wallets".